src/Microsoft.OpenApi.OData.Reader/Generator/OpenApiSchemaGenerator.cs

```diff
@@ -566,10 +566,21 @@ internal static OpenApiSchema CreateSchemaTypeSchema(this ODataContext context,
 
                 if (context.Settings.EnableDiscriminatorValue)
                 {
+                    OpenApiString defaultValue = null;
+                    bool isBaseTypeEntity = "entity".Equals(structuredType.BaseType?.FullTypeName().Split('.').Last(), StringComparison.OrdinalIgnoreCase);
+                    bool isBaseTypeAbstractNonEntity = (structuredType.BaseType?.IsAbstract ?? false) && !isBaseTypeEntity;
+
+                    if (context.Settings.EnableDefaultValueForOdataTypeProperty ||
```

Comment:
shouldn't this be && instead of || ??

Reply:
Yes, good catch! It should be:
```
if (context.Settings.EnableDefaultValueForOdataTypeProperty ||
    (!context.Settings.EnableDefaultValueForOdataTypeProperty &&
    (isBaseTypeAbstractNonEntity ||   
  context.Model.IsBaseTypeReferencedAsTypeInModel(structuredType.BaseType))))
```

Reply:
I believe it should in fact be 
```CSharp
if (context.Settings.EnableDefaultValueForOdataTypeProperty && (isBaseTypeAbstractNonEntity || context.Model.IsBaseTypeReferencedAsTypeInModel(structuredType.BaseType)))
```
since you always want the abstraction aspect to be considered, not just the setting itself. Does that make sense?

Reply:
In my head, we are trying to a solve a problem specific to graph as the existence of the property by default should not really be a problem for an OData API. 

Currently, the setting therefore creates the possibility of disabling this "custom" behavior, by enabling all models to have the property set by default, otherwise having them only set when type disambiguation is really needed. 

The proposed change would have the setting either have the property set only when type disambiguation is really needed or not at all. Which still works as we can use the setting to get what we want. 

I think the proposal by @irvinesunday makes sense with a possible renaming of the setting to `EnableOdataTypePropertyOnlyForTypeDisambiguation` so that it makes it clearer in the context.

Reply:
aaah ok, thanks for clarifying here, I think I missed some of the intent. So we're still going to always add the odata type by default unless the disambiguation is on AND we match conditions when not to include it to avoid ambiguous situations.
On top of the setting rename, I'd like to suggest setting it to false by default.

The condition should then be something like

```CSharp
if (!context.Settings.EnableDefaultValueForOdataTypeProperty || (context.Settings.EnableDefaultValueForOdataTypeProperty && (isBaseTypeAbstractNonEntity || context.Model.IsBaseTypeReferencedAsTypeInModel(structuredType.BaseType))))
```
makes sense?

Reply:
Agreed!

Reply:
Renamed the convert setting to `EnableTypeDisambiguationForDefaultValueOfOdataTypeProperty` and simplified the condition to:
```
if (!context.Settings.EnableTypeDisambiguationForDefaultValueOfOdataTypeProperty ||
    isBaseTypeAbstractNonEntity ||                  
    context.Model.IsBaseTypeReferencedAsTypeInModel(structuredType.BaseType))
```

Reply:
right boolean algebra simplifications, this takes me back :)